core/modules/map.rs

```diff
@@ -1104,6 +1118,11 @@ impl ModuleMap {
             _ = sender.sender.take().unwrap().send(Err(err.into()));
           }
           PromiseState::Pending => {
+            // User code shouldn't be able to both cause the runtime to fail and leave the promise as
```

Comment:
i feel fairly confident about this but i'd love if other folks could confirm that this makes sense